Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -86,11 +86,28 @@ unsigned ScriptDisallowedScope::s_count = 0;
 ScriptDisallowedScope::EventAllowedScope* ScriptDisallowedScope::EventAllowedScope::s_currentScope = nullptr;
 #endif
 
-ALWAYS_INLINE auto ContainerNode::removeAllChildrenWithScriptAssertion(ChildChange::Source source, NodeVector& children, DeferChildrenChanged deferChildrenChanged) -> DidRemoveElements
+ALWAYS_INLINE auto ContainerNode::removeAllChildrenWithScriptAssertionMaybeAsync(ChildChange::Source source, NodeVector& children, DeferChildrenChanged deferChildrenChanged) -> RemoveAllChildrenResult
+{
+#if PLATFORM(MAC)
```

Comment:
Why does this code need to be Mac specific? We should at least explain why in a comment or in the commit message.

Reply:
Could you add a FIXME here saying that we should have a unified code path?